x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/ldap/PoolingSessionFactory.java

```diff
@@ -91,6 +91,17 @@
         if (bindDn == null) {
             bindCredentials = new SimpleBindRequest();
         } else {
+            if (bindPassword == null) {
+                throw new IllegalArgumentException(
+                    "When ["
+                        + RealmSettings.getFullSettingKey(config, BIND_DN)
+                        + "] is set you must also specify ["
+                        + RealmSettings.getFullSettingKey(config, LEGACY_BIND_PASSWORD)
+                        + "] or ["
+                        + RealmSettings.getFullSettingKey(config, SECURE_BIND_PASSWORD)
```

Comment:
We should list `SECURE_BIND_PASSWORD` before `LEGACY_BIND_PASSWORD` as it is the preferred (non deprecated) setting and people are more likely to follow the first suggested fix.